dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityAttributeStoreTest.java

```diff
@@ -144,6 +168,64 @@ public void testSaveTrackedEntityAttribute()
         assertNotNull( attributeService.getTrackedEntityAttribute( idB ) );
     }
 
+    @Test
+    public void testGetAllIndexableAttributes()
+    {
+        long idA = attributeService.addTrackedEntityAttribute( attributeW );
+        long idB = attributeService.addTrackedEntityAttribute( attributeY );
+
+        Set<TrackedEntityAttribute> indexableAttributes = attributeService
+            .getAllTrigramIndexableTrackedEntityAttributes();
+
+        assertNotNull( indexableAttributes );
+        assertEquals( indexableAttributes.size(), 9 );
+        assertTrue( indexableAttributes.contains( attributeW ) );
+        assertTrue( indexableAttributes.contains( attributeY ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'A' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'E' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'J' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'N' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'O' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'S' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'T' ) ) );
+    }
+
+    @Test
+    public void testCreateTrigramIndex()
+    {
+        long idA = attributeService.addTrackedEntityAttribute( attributeW );
+        Exception exception = assertThrows( BadSqlGrammarException.class,
+            () -> attributeService.createTrigramIndex( attributeW ) );
```

Comment:
why are we expecting this to throw?

Reply:
Good question. Because these unit tests run on H2 and the trigram indexing queries are incompatible, but I wanted to be sure that the generated sql is correct :D I guess I better mock the db call and simply make sure the correct sql is used to invoke it.

Reply:
Would it then make sense to come up with a test running on postgres? Only for the postgres specific things. Is there something you could test using an SQL statement to make sure the trigram index was added where you wanted it? So executing your code, then executing an SQL which would be the assertion 🤔

Reply:
Created a separate Integration Test and removed the corresponding tests from unit test class.